server/src/main/java/org/apache/iotdb/db/query/udf/core/transformer/ArithmeticBinaryTransformer.java

```diff
@@ -36,6 +36,11 @@ protected ArithmeticBinaryTransformer(
     this.rightPointReader = rightPointReader;
   }
 
+  @Override
+  public boolean isConstantPointReader() {
+    return leftPointReader.isConstantPointReader() && rightPointReader.isConstantPointReader();
```

Comment:
It's better to cache the result of isConstantPointReader as a class member of Expression. Otherwise, a cachetime() call will cause multiple chain isConstantPointReader() calls.

Reply:
Make sense. Will cache the result.